proxy/src/main/java/net/md_5/bungee/BungeeBossBar.java

```diff
@@ -0,0 +1,280 @@
+package net.md_5.bungee;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.EnumSet;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+import net.md_5.bungee.api.boss.BossBarColor;
+import net.md_5.bungee.api.boss.BossBarDivision;
+import net.md_5.bungee.api.boss.BossBarFlag;
+import net.md_5.bungee.api.chat.BaseComponent;
+import net.md_5.bungee.api.connection.ProxiedPlayer;
+import net.md_5.bungee.chat.ComponentSerializer;
+import net.md_5.bungee.protocol.DefinedPacket;
+import net.md_5.bungee.protocol.ProtocolConstants;
+import net.md_5.bungee.protocol.packet.BossBar;
+
+@ToString
+@EqualsAndHashCode
+public class BungeeBossBar implements net.md_5.bungee.api.boss.BossBar
+{
+
+    @Getter
+    private BaseComponent[] title;
+    @Getter
+    private BossBarColor color;
+    @Getter
+    private BossBarDivision division;
+    @Getter
+    private float health;
+    @Getter
+    private boolean visible;
+
+    private Set<ProxiedPlayer> players;
+    private EnumSet<BossBarFlag> flags;
+
+    @ToString.Exclude
+    private UUID uuid;
+
+    public BungeeBossBar(BaseComponent[] title, BossBarColor color, BossBarDivision division, float health)
+    {
+        this.title = Preconditions.checkNotNull( title, "title" );
+        this.color = Preconditions.checkNotNull( color, "color" );
+        this.division = Preconditions.checkNotNull( division, "division" );
+        Preconditions.checkArgument( 0 <= health && health <= 1, "Health may not be lower than 0 or greater than 1" );
+        this.health = health;
+        this.visible = true;
+        this.players = new HashSet<>();
+        this.flags = EnumSet.noneOf( BossBarFlag.class );
+        this.uuid = UUID.randomUUID();
+    }
+
+    @Override
+    public boolean canBeAdded(ProxiedPlayer player)
+    {
+        return player.getPendingConnection().getVersion() >= ProtocolConstants.MINECRAFT_1_9;
+    }
+
+    @Override
+    public void addPlayer(ProxiedPlayer player)
+    {
+        Preconditions.checkNotNull( player, "player" );
```

Comment:
Perhaps these methods should check canBeAdded ?

Reply:
Wdym?

Reply:
Will move that check there

Reply:
I don't think this is better.
I would either make it throw an exception or at least return a boolean.

Reply:
I don't understand what you mean. It's just a protocol version check. They can't create a bossbar and add the player somewhere where the protocol version isn't set. There won't be a exception either.

Reply:
The point is that the method just silently returns if the player is on an old version and the caller will never know the player wasn't actually added to the bar

Reply:
If a plugin cares about this, it can use the `canBeAdded` method.

Reply:
> The point is that the method just silently returns if the player is on an old version and the caller will never know the player wasn't actually added to the bar

If the caller cares, how Janmm said he could check with canBeAdded.